test/integration/conflict_resolution.js

```diff
@@ -192,6 +192,164 @@ describe('Conflict resolution', () => {
+      should(await helpers.trees()).deepEqual(expectedTree)
+    })
+
+    it('does not trigger a conflict on subsequent local update', async () => {
+      await helpers.local.syncDir.outputFile('same-name', 'content2')
+      await fullSyncStartingFrom('local')
+
+      should(await helpers.trees()).deepEqual(expectedTree)
+    })
+  })
+
+  describe('merging local dir then remote file', () => {
+    it('renames the remote file', async () => {
+      await helpers.local.syncDir.ensureDir('same-name')
+      await cozy.files.create('content2', {
+        name: 'same-name',
+        contentType: 'text/plain'
+      })
+
+      await fullSyncStartingFrom('local')
+
+      should(await helpers.trees()).deepEqual(bothSides([
+        'same-name-conflict-...',
+        'same-name/'
+      ]))
+    })
+  })
+
+  describe('merging local file addition then remote file move to the same destination', () => {
+    it('renames the remote move destination', async () => {
+      await helpers.local.syncDir.outputFile('src', 'src content')
+      await helpers.local.scan()
+      await helpers.syncAll()
+      // FIXME: Initial tree helper?
+      const remoteFile = await cozy.files.statByPath(`/src`)
+      await cozy.files.updateAttributesById(remoteFile._id, {name: 'dst'})
+      await helpers.local.syncDir.outputFile('dst', 'local dst content')
+
+      await fullSyncStartingFrom('local')
+
+      should(await helpers.trees()).deepEqual(bothSides([
+        'dst',
+        'dst-conflict-...'
+      ]))
+    })
+  })
+
+  describe('merging local file move then remote file addition', () => {
+    it('renames the remote file', async () => {
+      await helpers.local.syncDir.outputFile('src', 'initial content')
+      await helpers.local.scan()
+      await helpers.syncAll()
+      await helpers.pullAndSyncAll()
+      // FIXME: Initial tree helper?
+      await cozy.files.create('remote dst content', {name: 'dst'})
+      await helpers.local.syncDir.move('src', 'dst')
+
+      await fullSyncStartingFrom('local')
+
+      should(await helpers.trees()).deepEqual(bothSides([
+        'dst',
+        'dst-conflict-...'
+      ]))
+    })
+  })
+
+  describe('merging local dir addition then remote move to the same destination', () => {
+    it('renames the remote move destination', async () => {
+      await helpers.local.syncDir.ensureDir('src')
+      await helpers.local.scan()
+      await helpers.syncAll()
+      // FIXME: Initial tree helper?
+      const remoteFile = await cozy.files.statByPath(`/src`)
+      await cozy.files.updateAttributesById(remoteFile._id, {name: 'dst'})
+      await helpers.local.syncDir.ensureDir('dst')
+
+      await fullSyncStartingFrom('local')
+
+      should(await helpers.trees()).deepEqual(bothSides([
+        'dst-conflict-.../',
+        'dst/'
+      ]))
+    })
+  })
+
+  describe('merging local dir move then remote dir addition to the same destination', () => {
+    it('renames the remote dir', async () => {
```

Comment:
Should we really merge this test when the resulting situation is not the one we're actually expecting?

Reply:
I don't know, I have the feeling this does the opposite. Maybe I'm wrong though.

Reply:
Ok, I'll replace it with a notice and move it with the fix.